### DISCUSSION TITLE
how use object in FileElement

### DISCUSSION THREAD

**Discussion by hosseinjarrahi** [vueform/vueform#199] [Questions] [answered]
i use FileElement to upload a file, but when the file is uploaded, request will return an object as response that has a name and a string as url to show the file, in FileElement we can just set one thing as name and url , is it possible to set name and url diffrent?

**Comment by adamberecz** [comment]
So just to make sure I understand it correctly: you have uploaded and persisted a file ([stage 3](https://vueform.com/docs/file-uploads#image-upload-diagram)) and the `FileElement`'s value is eg. `blue-orange.png` or `https://domain.com/images/blue-orange.png` and instead of `blue-orange.png` you want to show something like `Blue Orange` for the file?
<img width="228" alt="Screenshot 2024-04-08 at 17 43 11" src="https://github.com/vueform/vueform/assets/16024069/02c85d71-b3f7-4b68-af7d-2e13a0c33528">

**Reply by hosseinjarrahi** [reply]
for example my response is like this
```js
{
name: 'blueOrange.jpg',
hash: 'Dasd8asd8cNCsajkscas'
}
```
and url for image view is
http://localhost:9090/images?hash=Dasd8asd8cNCsajkscas

but i have no idea how FileElement  can handle it

**Reply by adamberecz** [reply]
This response object with `name` and `hash` is in `uploadTempEndpoint` (so when initially uploaded in a single request before submitting the form) or when you want to load the image back to the form?

**Reply by hosseinjarrahi** [reply]
Thanks for answering, bro. Before submitting the form, when click to FileElement or MultiFileElement, select the file and after uploading, the response back from the server has two key, hash and name. The name is the name of file and with hash we can download the file. i dont know how show the name file with name property and when user click on the button, download The file using hash in this url http://localhost:9090/images?hash=Dasd8asd8cNCsajkscas

**Reply by adamberecz** [reply, marked as answer]
I wasn't able to achieve this myself. The issue is that currently you can only use the same filename for rendering preview, displaying name and handling click while being quite limited when it comes to formatting the urls.

I have to figure out a way to restructure the file upload to be able to achieve this. I've created a discussion for hit: https://github.com/vueform/vueform/discussions/239

**Comment by hosseinjarrahi** [comment]
no answer yet?

**Reply by adamberecz** [reply]
Apologies, I overlooked that there was a new activity on this discussion - answered you above.

**Comment by hosseinjarrahi** [comment]
🥲
